### PR TITLE
SystemConfig: show WSL version if available on Linux

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This should make it easier to determine issues with WSL kernels

### CODE REVIEW

**Review by dawidd6 — APPROVED** [review]

File: Library/Homebrew/extend/os/linux/system_config.rb [hunk 1]
```diff
@@ -42,8 +42,9 @@ def host_ruby_version
 
     def dump_verbose_config(out = $stdout)
       dump_generic_verbose_config(out)
-      out.puts "Kernel: #{`uname -mors`.chomp}"
+      out.puts "Kernel: #{Utils.safe_popen_read("uname", "-mors").chomp}"
       out.puts "OS: #{OS::Linux.os_version}"
+      out.puts "WSL: #{Utils.safe_popen_read("wsl", "--version").chomp}" if which("wsl")
```
Comment (Bo98) [inline Library/Homebrew/extend/os/linux/system_config.rb:47]: `wsl --version` is not available within WSL - it's only available in a Windows command prompt.

**Review by MikeMcQuaid — COMMENTED** [review]
Makes sense to me. Happy to ✅ after Bo98's comment addressed.

File: Library/Homebrew/extend/os/linux/system_config.rb [hunk 2]
```diff
@@ -40,10 +40,20 @@ def host_ruby_version
       out
     end
 
+    def wsl_version(kernel)
+      return "no" unless /-microsoft/i.match?(kernel)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/linux/system_config.rb:44]:
```suggestion
      return unless /-microsoft/i.match?(kernel)
```

Let's not print anything here on the typical, non-WSL case.

File: Library/Homebrew/extend/os/linux/system_config.rb [hunk 3]
```diff
@@ -40,10 +40,20 @@ def host_ruby_version
...
+      return "1" if /-Microsoft/.match?(kernel)
+    end
+
     def dump_verbose_config(out = $stdout)
+      kernel = Utils.safe_popen_read("uname", "-mors").chomp
       dump_generic_verbose_config(out)
-      out.puts "Kernel: #{`uname -mors`.chomp}"
+      out.puts "Kernel: #{kernel}"
       out.puts "OS: #{OS::Linux.os_version}"
+      out.puts "WSL: #{wsl_version(kernel)}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/linux/system_config.rb:56]:
```suggestion
      if (wsl = wsl_version(kernel).presence)
        out.puts "WSL: #{wsl}"
      end
```

Let's not print anything here on the typical, non-WSL case.

File: Library/Homebrew/extend/os/linux/system_config.rb [hunk 4]
```diff
@@ -40,10 +40,20 @@ def host_ruby_version
       out
     end
 
+    def wsl_version(kernel)
+      return "no" unless /-microsoft/i.match?(kernel)
+
+      return "2 (store)" if Version.new kernel[/Linux ([0-9.]*)-.*/, 1] > Version.new("5.15")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/linux/system_config.rb:46]:
```suggestion
      return "2 (Microsoft Store)" if Version.new(kernel[/Linux ([0-9.]*)-.*/, 1]) > Version.new("5.15")
```

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work! Some suggested tweaks.

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**